bc_obps/compliance/service/penalty_calculation_service.py

```diff
@@ -281,6 +379,22 @@ def calculate_penalty(
 
         return result
 
+    @classmethod
+    def _get_last_supplementary_submission_date(cls, obligation: ComplianceObligation) -> date | None:
```

Comment:
I don't think this function is necessary, we can just check the date of the obligation